src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1446,6 +1446,10 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
 
         $loader->load('messenger.xml');
 
+        if (!$container->getParameter('kernel.debug') || !$container->hasAlias('logger')) {
```

Comment:
Actually, the `!$container->hasAlias('logger')` part could be removed, as the Fwb always registers a default logger implementation since #24300

But anyway that'll be handled differently with #26864 as it'll have to be opt-in explicitly.

Reply:
AFAIK doing so will always remove the middleware. I don't see any value in changing what was just changed in #26896 :)

Reply:
The `!$container->hasAlias('logger')` check in the fwb extension will indeed lead to the middleware removal, even if there is in fact a `logger` alias declared by the monolog bundle or the kernel `LoggerPass`, so it's wrong. But as stated above, this check is not necessary anyway, as there will always be a `logger` alias.

Regarding #26896 , I agree with @yceruto: It doesn't really make sense to me having a LoggingMiddleware without a hard logger dependency. It solves an issue we usually solve at compilation time by removing proper services.

Reply:
I'll try this again soon.